CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -900,8 +900,17 @@ public static function calculateRecurLineItems($recurId, $total_amount, $financi
     $originalContributionID = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution', $recurId, 'id', 'contribution_recur_id');
     $lineItems = CRM_Price_BAO_LineItem::getLineItemsByContributionID($originalContributionID);
     $lineSets = array();
+    $contributionById = civicrm_api3('Contribution', 'getsingle', array(
+    'id' => $originalContributionID,
+    ));
+
```

Comment:
@Elennio this could be further optimised into 
```patch
-   $originalContributionID = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution', $recurId, 'id', 'contribution_recur_id');
-   $lineItems = CRM_Price_BAO_LineItem::getLineItemsByContributionID($originalContributionID);
+   $originalContribution = civicrm_api3('Contribution', 'getsingle', array('id' => $originalContributionID));
+   $lineItems = CRM_Price_BAO_LineItem::getLineItemsByContributionID($originalContribution['id']);
       $lineSets = array();
  ```
  What do you think?  :)

Reply:
I think optimize its the way to go, so I will do.